pilot/pkg/networking/core/v1alpha3/waypoint.go

```diff
@@ -48,16 +48,37 @@ type waypointServices struct {
 func findWaypointResources(node *model.Proxy, push *model.PushContext) ([]model.WorkloadInfo, *waypointServices) {
```

Comment:
There are some callers that only care about either services OR waypoints.Maybe we should split this into `findWaypointServices` and `findWaypointWorkloads`?

Reply:
I don't see right off hand anything that only cares about workloads. It seems like common use is either want both or want only services. I don't feel super strongly about it really but in the service only case I'm not that certain it's overly burdensome to do the indexed wl lookup. I can see where if we commonly wanted just workloads how it would be an improvement to not do the service-specific work but at the moment I don't see that use. Is something you are working on changing things to where wl-only lookups will be common?

It seems simple enough to break these out into their own separate lookups if that's strongly preferred though. Like I said I don't feel that strongly either way. LMK what you are thinking

Reply:
This should have been a nit. I don't feel strongly here.

Reply:
I like the way you handled it in the sandwich PR too. Its tidy. Will not be offended if we stick with your impl when sandwich merges.